lib/database/auth.php

```diff
@@ -89,113 +89,42 @@ function validateUser_app(&$user, $token, &$fbUser, $permissionRequired): bool
     );
 }
 
-function validateUser_cookie(&$user, $cookie, $permissionRequired, &$permissions = 0): bool
-{
-    return validateFromCookie($user, $points, $permissions, $permissionRequired);
-}
-
-function validateFromCookie(&$userOut, &$pointsOut, &$permissionsOut, $permissionRequired = 0): bool
```

Comment:
`validateUser_cookie` and `validateFromCookie` were effectively the same as `RA_ReadCookieCredentials`, with different parameters. All three have been merged into a singular `RA_ValidateCookie`.

Reply:
I like those changes a lot! 

I'd like to rename those `RA_` prefixed functions to be more in line with the rest:
- `ValidateCookie`
- `ValidateToken` (dropping the redundant `Credentials`)
- `ClearCookie`
- `ReadCookie`
- `ApplyCookie` (`storecookie` already exists in php)
- `CookieExists`

`RA_ReadTokenCredentials` looks a lot like `RA_ValidateCookie` used to, where `points`, `truePoints`, and `unreadMessages` are passed by reference. Yet it seems those aren't set in that function nor used as such in `dorequrest` nor `doupload`. Those parameters can be removed if I'm not mistaken.

I can add those changes as an additional commit or PR even - what do you think?

Reply:
The more I look the more i find ... `validateUser()` still has a `fbUser` parameter, 
`validateUser_app()` isn't used anymore and can be removed, too.

Reply:
`authenticateWith...` instead of `validate...` even?

Reply:
> I can add those changes as an additional commit or PR even - what do you think?

This PR is already touching all the files, so we should get the naming correct here to avoid another PR touching all the files.

You want me to change `RA_ValidateCookie` to `authenticateWithCookie`?
```
authenticateWithCookie($user, $permissions, $userDetails)
```
and
```
if (!authenticateWithCookie($user, $permissions, $userDetails, Permissions::Developer)) {
```

What do you think about `getUserFromCookie`?

Reply:
> I can add those changes as an additional commit or PR even - what do you think?

I guess you're asking two separate questions here.

I think the `RA_ValidateCookie` name should be finalized in this PR. the changes to clean up `RA_ReadTokenCredentials`, `validateUser`, and `validateUser_app` should be in a separate PR.

Reply:
> I think the RA_ValidateCookie name should be finalized in this PR. the changes to clean up RA_ReadTokenCredentials, validateUser, and validateUser_app should be in a separate PR.

That sounds good 👍 

> What do you think about `getUserFromCookie`?

I think I'd prefer either `validate` or `authenticateWith` as getting it from cookie sounds like user information is stored within cookies/token/password. The reasoning for not using `validate` is that it sounds like validating a value by rules whereas it's a full authentication/authorization routine.

At the end of the day it doesn't really matter that much to me :) I'm fine with whatever you prefer here.

Reply:
I'm torn because the function is overloaded to authenticate _and_ return the user data. The only actual input parameter is the optional minimum permissions level.

I'm going to go with `authenticateFromCookie`. For some reason, the "with" just doesn't sit well with me.

Reply:
I like that :) I wasn't too happy using `with` either 👍